server.js

```diff
@@ -1,248 +1,229 @@
+  "View all roles": viewAllRoles,
+  "View all employees": viewAllEmployees,
+  "Add a department": addADepartment,
+  "Add a role": addARole,
+  "Add an employee": addAnEmployee,
+  "Update employee roles": updateEmployeeRoles,
+  "Exit program": exitProgram,
+};
 
 const connection = mysql.createConnection({
-    host: 'localhost',
-    port: 3306,
-    user: 'root',
-    password: '',
-    database: 'employeeTrackerDB',
+  host: 'localhost',
+  port: 3306,
+  user: 'root',
+  password: '',
+  database: 'employeeTrackerDB',
 });
 
 connection.connect((err) => {
-    if (err) throw err;
-    console.log(`Connected as id ${connection.threadId} \n`);
-    startApp();
+  if (err) throw err;
+  console.log(`Connected as id ${connection.threadId} \n`);
+  startApp();
 });
 
-startApp = () => {
-    inquirer.prompt([
-        {
-            name: 'initialInquiry',
-            type: 'rawlist',
-            message: 'Welcome to the employee management program. What would you like to do?',
-            choices: ['View all departments', 'View all roles', 'View all employees', 'Add a department', 'Add a role', 'Add an employee', 'Update employee roles', 'Exit program']
-        }
-    ]).then((response) => {
-        switch (response.initialInquiry) {
-            case 'View all departments':
-                viewAllDepartments();    
-                break;
-            case 'View all roles':
-                viewAllRoles();
-                break;
-            case 'View all employees':
-                viewAllEmployees();
-                break;
-            case 'Add a department':
-                addADepartment();
-            break;
-            case 'Add a role':
-                addARole();
-            break;
-            case 'Add an employee':
-                addAnEmployee();
-            break;
-            case 'Update employee roles':
-                updateEmployeeRole();
-            break;
-            case 'Exit program':
-                connection.end();
-                console.log('\n You have exited the employee management program. Thanks for using! \n');
-                return;
-            default:
-                break;
-        }
-    })
+function startApp() {
+  inquirer.prompt([
+    {
+      name: 'initialInquiry',
+      type: 'rawlist',
+      message: 'Welcome to the employee management program. What would you like to do?',
+      choices: Object.keys(actions)
+    }
+  ]).then((response) => {
+    const actionToExecute = actions[actionName];
+    if (actionToExecute) {
+      actionToExecute();
+    }
+
+    if (!programIsRunning) {
+      return;
+    }
+  })
 }
 
-viewAllDepartments = () => {
-    connection.query(`SELECT * FROM department ORDER BY department_id ASC;`, (err, res) => {
-        if (err) throw err;
-        console.table('\n', res, '\n');
-        startApp();
+function viewAllDepartments() {
+  connection.query(`SELECT * FROM department ORDER BY department_id ASC;`)
+    .then(res => {
```

Comment:
@samrogers15 This is a simple conversion from the callbacks style of writing async code to .then syntax, but I would recommend using this everywhere. Alternatively, you could use `async/await`, but I honestly prefer the error handling with .then chaining since you're able to write all your code with as many .thens as you need and taper it off with a single .catch statement at the end. I saw you had a couple .thens already in your code, so I assume you're already familiar with how it works, but if not you can [check this out](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then)